test/pkg/compose/build/build_blueprint.go

```diff
@@ -0,0 +1,279 @@
+		return nil, fmt.Errorf("failed to obtain value of field %q in file %q", "name", path)
+	}
+
+	// If name has contains templating syntax, it needs to be templated to get true name
+	if strings.Contains(name, "{{") {
+		name, err = opts.TplData.Template(fmt.Sprintf("name-of-%s", filename), name)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	templatedData, err := opts.TplData.Template(name, data)
+	if err != nil {
+		return nil, err
+	}
+
+	bb := &BlueprintBuild{
+		build: build{
+			Name: name,
+			Path: path,
+		},
+		Contents: templatedData,
+	}
+
+	// blueprint.alias file contains aliases for commit defined in blueprint.toml
+	potentialAliasFile := fmt.Sprintf("%s.alias", withoutExt)
+	if exists, err := fileExistsInDir(dir, potentialAliasFile); err != nil {
+		return nil, err
+	} else if exists {
+		data, err := os.ReadFile(filepath.Join(dir, potentialAliasFile))
+		if err != nil {
+			return nil, err
+		}
+		bb.Aliases = slices.DeleteFunc(strings.Split(string(data), "\n"), func(line string) bool { return line == "" })
+	}
+
+	if opts.BuildInstallers {
+		// If blueprint.image-installer exists, then ISO installer can be built.
+		if exists, err := fileExistsInDir(dir, fmt.Sprintf("%s.image-installer", withoutExt)); err != nil {
+			return nil, err
+		} else if exists {
+			bb.Installer = true
+			bb.InstallerDestination = filepath.Join(opts.ArtifactsMainDir, "vm-storage", name+".iso")
+		}
+	}
+
+	// looking for parent commit
+	if strings.Contains(withoutExt, "-") {
+		parts := strings.Split(withoutExt, "-")
+		expectedParentFilename := parts[0] + ".toml"
+
+		parentPath := ""
+		err = filepath.WalkDir(filepath.Join(opts.TestDir, "image-blueprints"), func(p string, d fs.DirEntry, err error) error {
+			if parentPath != "" {
+				return nil
+			}
+			if d.Name() == expectedParentFilename {
+				parentPath = p
+			}
+			return nil
+		})
+		if err != nil {
+			return nil, fmt.Errorf("walking through dirs to find parent of %q failed: %w", path, err)
+		}
+
+		if parentPath != "" {
+			parentData, err := os.ReadFile(parentPath)
+			if err != nil {
+				return nil, fmt.Errorf("failed to read parent of %q which is %q: %w", path, parentPath, err)
+			}
+			parentName, err := testutil.GetTOMLFieldValue(string(parentData), "name")
+			if err != nil {
+				return nil, fmt.Errorf("failed to read name from %q: %w", parentPath, err)
+			}
+			bb.Parent = parentName
+		}
+	}
+
+	return bb, nil
+}
+
+func (b *BlueprintBuild) Prepare(opts *Opts) error {
+	// TODO: Do we need to remove Blueprint before to adding? It's not required and it only bumps blueprint's version
+	err := opts.Composer.AddBlueprint(b.Contents)
+	if err != nil {
+		return err
+	}
+	err = opts.Composer.DepsolveBlueprint(b.Name)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (b *BlueprintBuild) Execute(opts *Opts) error {
+	refExists, err := opts.Ostree.DoesRefExists(b.Name)
+	if err != nil {
+		return err
+	}
+
+	eg, _ := errgroup.WithContext(context.TODO())
```

Comment:
Maybe pass a context in to Execute() to support cancelling?

Reply:
This could be part of enhancement to cancel the builds upon ctrl+C, right now nothing even honors context. And we'd have to roll out our own WaitForCompose function because existing only takes internal and timeout duration.

Reply:
Sure, yeah, I wouldn't hold this PR up over it.